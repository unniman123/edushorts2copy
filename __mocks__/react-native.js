const React = require('react');

const mockNativeModules = {
  SettingsManager: {
    settings: {
      AppleLocale: 'en_US',
    },
    getConstants: () => ({
      settings: {
        AppleLocale: 'en_US',
      },
    }),
  },
  BatteryManager: {
    getTemperature: jest.fn().mockResolvedValue(25),
  },
  DeviceInfo: {
    getConstants: () => ({
      Dimensions: {
        window: {
          width: 375,
          height: 812,
          scale: 2,
          fontScale: 1,
        },
      },
    }),
  },
};

const mockEventEmitter = {
  addListener: jest.fn(),
  removeAllListeners: jest.fn(),
  removeSubscription: jest.fn(),
  listeners: jest.fn(() => []),
  removeListener: jest.fn(),
};

const RN = {
  Platform: {
    OS: 'ios',
    select: jest.fn(options => options.ios),
  },
  StyleSheet: {
    create: jest.fn(styles => styles),
    hairlineWidth: 1,
  },
  View: 'View',
  Text: 'Text',
  TouchableOpacity: 'TouchableOpacity',
  ScrollView: 'ScrollView',
  Animated: {
    View: 'Animated.View',
    timing: jest.fn().mockReturnValue({
      start: jest.fn(callback => callback && callback()),
    }),
    Value: jest.fn(() => ({
      interpolate: jest.fn(),
      setValue: jest.fn(),
    })),
  },
  Dimensions: {
    get: jest.fn().mockReturnValue({ width: 375, height: 812 }),
  },
  NativeModules: mockNativeModules,
  NativeEventEmitter: jest.fn(() => mockEventEmitter),
  InteractionManager: {
    runAfterInteractions: jest.fn(callback => callback()),
  },
  Settings: {
    get: jest.fn(),
    set: jest.fn(),
  },
  TurboModuleRegistry: {
    getEnforcing: jest.fn((moduleName) => {
      if (moduleName === 'SettingsManager') {
        return mockNativeModules.SettingsManager;
      }
      return {};
    }),
  },
  ProgressBarAndroid: 'ProgressBarAndroid',
  Clipboard: {
    getString: jest.fn(),
    setString: jest.fn(),
  },
  PushNotificationIOS: {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    requestPermissions: jest.fn(),
  },
  DevSettings: {
    addMenuItem: jest.fn(),
    reload: jest.fn(),
  },
  VirtualizedList: 'VirtualizedList',
};

module.exports = RN; 