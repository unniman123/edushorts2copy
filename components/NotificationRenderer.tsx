import React, { useCallback } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Platform,
  ViewStyle,
  TextStyle,
} from 'react-native';
import { useNavigation, NavigationProp } from '@react-navigation/native';
import { RootStackParamList } from '../types/navigation';
import { Ionicons } from '@expo/vector-icons';
import { formatDistanceToNow } from 'date-fns';

interface NotificationData {
  id: string;
  title: string;
  body: string;
  timestamp: Date;
  deep_link?: string;
  read: boolean;
  type: 'push' | 'web' | 'scheduled' | 'article_link';
  data?: Record<string, unknown>;
}

interface NotificationGroupProps {
  date: string;
  notifications: NotificationData[];
  onPress: (notification: NotificationData) => void;
  onDismiss: (notification: NotificationData) => void;
}

interface NotificationRendererProps {
  notification: NotificationData;
  onPress?: (notification: NotificationData) => void;
  onDismiss?: (notification: NotificationData) => void;
  style?: ViewStyle;
}

const NotificationGroup: React.FC<NotificationGroupProps> = ({
  date,
  notifications,
  onPress,
  onDismiss,
}) => {
  return (
    <View style={styles.group}>
      <Text style={styles.groupDate}>{date}</Text>
      {notifications.map((notification) => (
        <NotificationRenderer
          key={notification.id}
          notification={notification}
          onPress={onPress}
          onDismiss={onDismiss}
        />
      ))}
    </View>
  );
};

const NotificationContent: React.FC<{ notification: NotificationData }> = ({
  notification,
}) => {
  return (
    <View style={styles.content}>
      <Text style={[styles.title, !notification.read && styles.unread]}>
        {notification.title}
      </Text>
      <Text style={styles.body} numberOfLines={2}>
        {notification.body}
      </Text>
      <Text testID="notification-timestamp" style={styles.time}>
        {formatDistanceToNow(new Date(notification.timestamp), { addSuffix: true })}
      </Text>
    </View>
  );
};

const NotificationRenderer: React.FC<NotificationRendererProps> = ({
  notification,
  onPress,
  onDismiss,
  style,
}) => {
  const navigation = useNavigation<NavigationProp<RootStackParamList>>();

  const handlePress = useCallback(() => {
    if (notification.deep_link && notification.deep_link.startsWith('edushorts://articles/')) {
      const articleId = notification.deep_link.split('/').pop();
      if (articleId) {
        navigation.navigate('ArticleDetail', { articleId });
      }
    }
    onPress?.(notification);
  }, [notification, onPress, navigation]);

  const handleDismiss = useCallback(() => {
    onDismiss?.(notification);
  }, [notification, onDismiss]);

  const getIconName = useCallback(() => {
    switch (notification.type) {
      case 'article_link':
        return 'document-text';
      case 'scheduled':
        return 'time';
      case 'web':
        return 'globe';
      default:
        return 'notifications';
    }
  }, [notification.type]);

  return (
    <TouchableOpacity
      testID="notification-container"
      style={[styles.container, !notification.read && styles.unreadContainer, style]}
      onPress={handlePress}
      activeOpacity={0.7}
    >
      <View style={styles.iconContainer}>
        <Ionicons
          testID="notification-icon"
          name={getIconName() as any}
          size={24}
          color="#007AFF"
        />
      </View>
      <NotificationContent notification={notification} />
      <TouchableOpacity
        testID="dismiss-button"
        style={styles.dismissButton}
        onPress={handleDismiss}
      >
        <Ionicons name="close" size={20} color="#999" />
      </TouchableOpacity>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 12,
    backgroundColor: '#fff',
    borderRadius: 8,
    marginVertical: 4,
    marginHorizontal: 16,
    ...Platform.select({
      ios: {
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.2,
        shadowRadius: 2,
      },
      android: {
        elevation: 2,
      },
    }),
  },
  unreadContainer: {
    backgroundColor: '#f0f9ff',
  },
  iconContainer: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: '#e6f3ff',
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 12,
  },
  content: {
    flex: 1,
    marginRight: 8,
  },
  title: {
    fontSize: 16,
    fontWeight: '500',
    marginBottom: 4,
    color: '#1a1a1a',
  },
  unread: {
    fontWeight: '600',
    color: '#000',
  },
  body: {
    fontSize: 14,
    color: '#666',
    marginBottom: 4,
  },
  time: {
    fontSize: 12,
    color: '#999',
  },
  dismissButton: {
    padding: 4,
    justifyContent: 'center',
    alignItems: 'center',
  },
  group: {
    marginBottom: 16,
  },
  groupDate: {
    fontSize: 14,
    fontWeight: '600',
    color: '#666',
    marginLeft: 16,
    marginBottom: 8,
  },
});

export type { NotificationData, NotificationRendererProps };
export { NotificationGroup };
export default NotificationRenderer;
