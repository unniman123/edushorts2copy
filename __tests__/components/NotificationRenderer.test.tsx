import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import NotificationRenderer, { NotificationData } from '../../components/NotificationRenderer';

// Mock navigation
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

// Mock Ionicons
jest.mock('@expo/vector-icons', () => ({
  Ionicons: 'Ionicons',
}));

describe('NotificationRenderer', () => {
  const mockNotification: NotificationData = {
    id: 'test-id',
    title: 'Test Notification',
    body: 'This is a test notification',
    timestamp: new Date(),
    read: false,
    type: 'push',
  };

  const mockPress = jest.fn();
  const mockDismiss = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders notification content correctly', () => {
    const { getByText } = render(
      <NotificationRenderer
        notification={mockNotification}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    expect(getByText(mockNotification.title)).toBeTruthy();
    expect(getByText(mockNotification.body)).toBeTruthy();
  });

  it('handles press events', () => {
    const { getByTestId } = render(
      <NotificationRenderer
        notification={mockNotification}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    fireEvent.press(getByTestId('notification-container'));
    expect(mockPress).toHaveBeenCalledWith(mockNotification);
  });

  it('handles dismiss events', () => {
    const { getByTestId } = render(
      <NotificationRenderer
        notification={mockNotification}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    fireEvent.press(getByTestId('dismiss-button'));
    expect(mockDismiss).toHaveBeenCalledWith(mockNotification);
  });

  it('navigates to article detail when deep link is present', () => {
    const notificationWithDeepLink: NotificationData = {
      ...mockNotification,
      deep_link: 'edushorts://articles/123',
    };

    const { getByTestId } = render(
      <NotificationRenderer
        notification={notificationWithDeepLink}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    fireEvent.press(getByTestId('notification-container'));
    
    expect(mockNavigate).toHaveBeenCalledWith('ArticleDetail', { articleId: '123' });
    expect(mockPress).toHaveBeenCalledWith(notificationWithDeepLink);
  });

  it('displays different icons based on notification type', () => {
    const types: NotificationData['type'][] = ['push', 'web', 'scheduled', 'article_link'];
    const expectedIcons = ['notifications', 'globe', 'time', 'document-text'];

    types.forEach((type, index) => {
      const { getByTestId } = render(
        <NotificationRenderer
          notification={{ ...mockNotification, type }}
          onPress={mockPress}
          onDismiss={mockDismiss}
        />
      );

      const icon = getByTestId('notification-icon');
      expect(icon.props.name).toBe(expectedIcons[index]);
    });
  });

  it('applies unread styles when notification is not read', () => {
    const { getByTestId } = render(
      <NotificationRenderer
        notification={{ ...mockNotification, read: false }}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    const container = getByTestId('notification-container');
    expect(container.props.style).toContainEqual(
      expect.objectContaining({ backgroundColor: '#f0f9ff' })
    );
  });

  it('formats timestamp correctly', () => {
    const now = new Date();
    const notification = {
      ...mockNotification,
      timestamp: now,
    };

    const { getByTestId } = render(
      <NotificationRenderer
        notification={notification}
        onPress={mockPress}
        onDismiss={mockDismiss}
      />
    );

    const timestamp = getByTestId('notification-timestamp');
    expect(timestamp.props.children).toContain('less than a minute ago');
  });
});
