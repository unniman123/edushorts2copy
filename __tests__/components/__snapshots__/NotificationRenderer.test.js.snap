// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationRenderer Component renders NotificationGroup correctly 1`] = `
<View
  style={
    {
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 14,
        "fontWeight": "600",
        "marginBottom": 8,
        "marginLeft": 16,
      }
    }
  >
    Today
  </Text>
  <TouchableOpacity
    activeOpacity={0.7}
    onPress={[Function]}
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 4,
          "padding": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
        },
        {
          "backgroundColor": "#f0f9ff",
        },
        undefined,
      ]
    }
    testID="notification-container"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f3ff",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 40,
        }
      }
    >
      <Ionicons
        color="#007AFF"
        name="notifications"
        size={24}
        testID="notification-icon"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "marginRight": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1a1a1a",
              "fontSize": 16,
              "fontWeight": "500",
              "marginBottom": 4,
            },
            {
              "color": "#000",
              "fontWeight": "600",
            },
          ]
        }
      >
        Test Notification
      </Text>
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "marginBottom": 4,
          }
        }
      >
        This is a test notification body
      </Text>
      <Text
        style={
          {
            "color": "#999",
            "fontSize": 12,
          }
        }
        testID="notification-timestamp"
      >
        5 minutes ago
      </Text>
    </View>
    <TouchableOpacity
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 4,
        }
      }
      testID="dismiss-button"
    >
      <Ionicons
        color="#999"
        name="close"
        size={20}
      />
    </TouchableOpacity>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.7}
    onPress={[Function]}
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderRadius": 8,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 4,
          "padding": 12,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 2,
        },
        {
          "backgroundColor": "#f0f9ff",
        },
        undefined,
      ]
    }
    testID="notification-container"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f3ff",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "marginRight": 12,
          "width": 40,
        }
      }
    >
      <Ionicons
        color="#007AFF"
        name="notifications"
        size={24}
        testID="notification-icon"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "marginRight": 8,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#1a1a1a",
              "fontSize": 16,
              "fontWeight": "500",
              "marginBottom": 4,
            },
            {
              "color": "#000",
              "fontWeight": "600",
            },
          ]
        }
      >
        Another Notification
      </Text>
      <Text
        numberOfLines={2}
        style={
          {
            "color": "#666",
            "fontSize": 14,
            "marginBottom": 4,
          }
        }
      >
        This is a test notification body
      </Text>
      <Text
        style={
          {
            "color": "#999",
            "fontSize": 12,
          }
        }
        testID="notification-timestamp"
      >
        5 minutes ago
      </Text>
    </View>
    <TouchableOpacity
      onPress={[Function]}
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 4,
        }
      }
      testID="dismiss-button"
    >
      <Ionicons
        color="#999"
        name="close"
        size={20}
      />
    </TouchableOpacity>
  </TouchableOpacity>
</View>
`;

exports[`NotificationRenderer Component renders correctly when notification is read 1`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      false,
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="notifications"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          false,
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`NotificationRenderer Component renders correctly with all props 1`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#f0f9ff",
      },
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="notifications"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "color": "#000",
            "fontWeight": "600",
          },
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`NotificationRenderer Component renders correctly with different notification types 1`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#f0f9ff",
      },
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="document-text"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "color": "#000",
            "fontWeight": "600",
          },
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`NotificationRenderer Component renders correctly with different notification types 2`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#f0f9ff",
      },
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="time"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "color": "#000",
            "fontWeight": "600",
          },
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`NotificationRenderer Component renders correctly with different notification types 3`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#f0f9ff",
      },
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="globe"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "color": "#000",
            "fontWeight": "600",
          },
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`NotificationRenderer Component renders correctly with different notification types 4`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
  style={
    [
      {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "marginHorizontal": 16,
        "marginVertical": 4,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      },
      {
        "backgroundColor": "#f0f9ff",
      },
      undefined,
    ]
  }
  testID="notification-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e6f3ff",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginRight": 12,
        "width": 40,
      }
    }
  >
    <Ionicons
      color="#007AFF"
      name="notifications"
      size={24}
      testID="notification-icon"
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "marginRight": 8,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#1a1a1a",
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 4,
          },
          {
            "color": "#000",
            "fontWeight": "600",
          },
        ]
      }
    >
      Test Notification
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#666",
          "fontSize": 14,
          "marginBottom": 4,
        }
      }
    >
      This is a test notification body
    </Text>
    <Text
      style={
        {
          "color": "#999",
          "fontSize": 12,
        }
      }
      testID="notification-timestamp"
    >
      5 minutes ago
    </Text>
  </View>
  <TouchableOpacity
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 4,
      }
    }
    testID="dismiss-button"
  >
    <Ionicons
      color="#999"
      name="close"
      size={20}
    />
  </TouchableOpacity>
</TouchableOpacity>
`;
